helm/odigos/templates/crds/odigos.io_nodedetailses.yaml

```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.19.0
+  labels:
+    odigos.io/system-object: "true"
+  name: nodedetailses.odigos.io
+spec:
+  group: odigos.io
+  names:
+    kind: NodeDetails
+    listKind: NodeDetailsList
+    plural: nodedetailses
+    singular: nodedetails
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: NodeDetails is the Schema for the nodedetails API
+        properties:
+          apiVersion:
+            description: |-
+              APIVersion defines the versioned schema of this representation of an object.
+              Servers should convert recognized schemas to the latest internal value, and
+              may reject unrecognized values.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            type: string
+          kind:
+            description: |-
+              Kind is a string value representing the REST resource this object represents.
+              Servers may infer this from the endpoint the client submits requests to.
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: NodeDetailsSpec defines the desired state of NodeDetails
+            properties:
+              cpuCapacity:
+                description: CPUCapacity is the CPU capacity of the node (number of
+                  cores)
+                type: integer
```

Comment:
I think Kubernetes has a built in type for resources like cpu or memory maybe worth using the same type used in Pod spec